### ISSUE TITLE
inconsistency of LEEP test return with reported scores in your paper and LEEP paper

### ISSUE THREAD

**Issue by infinity2357** [EvZissel/expgen#2]
Hello,

I used your implementation of LEEP and tried to test it on a few procgen environment to replicate the result but my results have a large deviation from what you have reported (and also the original LEEP paper).

For example these are what I got from running PPO and LEEP and maze environment for 50e6 environment step:
![plot_maze_seed_5](https://github.com/user-attachments/assets/45d8a497-239c-47b3-b724-75d70b79a388)
 
As you can see the test results are very different. I used these commands:
`python train_ppo.py --env-name maze --seed 5 --num-env-steps 50000000 --use_backgrounds`
`python train_LEEP.py --env-name maze --seed 5 --num-env-steps 50000000 --use_backgrounds`

And used the data in `progress_maze_seed_5.csv` in PPO and LEEP folder to plot it.
[ppo_progress_maze_seed_5.csv](https://github.com/user-attachments/files/17529175/ppo_progress_maze_seed_5.csv)
[leep_progress_maze_seed_5.csv](https://github.com/user-attachments/files/17529176/leep_progress_maze_seed_5.csv)

I also average the mean and got these results:
`PPO Test Mean Reward for maze: 5.47`
`LEEP Test Mean Reward for maze: 4.63`

And what you have reported in Section E of appendices `Results after convergence` in `Table 9` was:
`PPO Test Mean Reward for maze: 5.78 ± 0.39`
`LEEP Test Mean Reward for maze: 6.78 ± 0.21`

As you can see there is about 2 points divergence in the results that I got on 50e6 environment step and yours. I used only one seed `5` but in the range of std that you reported it is also unacceptable.

I couldn't find out the reason of this inconsistency and hope that you could help.

Kind regards,
Arash

**Comment by EvZissel** [comment]
Hi Arash, thank you for reaching out.
This implementation of LEEP is based on the original authors' implementation following our mutual correspondence.
The authors haven't made their official implementation public, and together with them, we attempted to retrace the exact hyperparameters they had used. Eventually, we couldn't quite attain their performance even with their help, hence we resorted to using their results as reported in the LEEP paper (and we mentioned this in our paper - see footnote 5 on page 8).
